modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1578,6 +1559,61 @@ export class Wallet implements IWallet {
     return this.bitgo.post(url).send({ shareOptions: params }).result();
   }
 
+  async prepareSharedKeychain(
+    walletPassphrase: string | undefined,
+    pubkey: string,
+    path: string
+  ): Promise<SharedKeyChain> {
+    let sharedKeychain: SharedKeyChain = {};
+
+    try {
+      const keychain = await this.getEncryptedUserKeychain();
+
+      // Decrypt the user key with a passphrase
+      if (keychain.encryptedPrv) {
+        if (!walletPassphrase) {
+          throw new Error('Missing walletPassphrase argument');
+        }
+
+        const userPrv = decryptKeychainPrivateKey(this.bitgo, keychain, walletPassphrase);
+        if (!userPrv) {
+          throw new Error('Unable to decrypt user keychain');
+        }
+
+        keychain.prv = userPrv;
+        const eckey = makeRandomKey();
+        const secret = getSharedSecret(eckey, Buffer.from(pubkey, 'hex')).toString('hex');
+        const newEncryptedPrv = this.bitgo.encrypt({ password: secret, input: keychain.prv });
+
+        // Only one of pub/commonPub/commonKeychain should be present in the keychain
+        let pub = keychain.pub ?? keychain.commonPub;
+        if (keychain.commonKeychain) {
+          pub =
+            this.baseCoin.getMPCAlgorithm() === 'eddsa'
+              ? EddsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain)
+              : EcdsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain);
+        }
+
+        sharedKeychain = {
+          pub,
+          encryptedPrv: newEncryptedPrv,
+          fromPubKey: eckey.publicKey.toString('hex'),
+          toPubKey: pubkey,
+          path: path,
+        };
+      }
+    } catch (e) {
+      if (e.message === 'No encrypted keychains on this wallet.') {
```

Comment:
I think this is old code but we really shouldn't be doing production logics based on the error messages. 
It would be fine in the tests to verify the error messages are as expected but not in the production code to base the logics on.
Please add an error class of `MissingEncryptedKeychainError extends Error` and do the logic with `instanceof`
```suggestion
      if (e instanceof MissingEncryptedKeychainError) {
```